### PR TITLE
[MI-2291] Add feature to show error panel when user is not connected

### PR DESCRIPTION
**Pull request by ayusht2810**
image

### CODE REVIEW

File: webapp/src/containers/addOrViewComments/index.tsx [hunk 1]
```diff
@@ -62,8 +63,8 @@ const AddOrViewComments = () => {
         }
 
         const payload: CommentsPayload = {
-            record_type: pluginState.openCommentModalReducer.recordType,
-            record_id: pluginState.openCommentModalReducer.recordId,
+            record_type: pluginState.openCommentModalReducer.data?.recordType,
+            record_id: pluginState.openCommentModalReducer.data?.recordId,
```
Comment (muazzamqazi) [inline webapp/src/containers/addOrViewComments/index.tsx:67]:
```suggestion
const payload: CommentsPayload = getCommentsPayload();
```
```
const getCommentsPayload = (): CommentsPayload => ({
  record_type: pluginState.openCommentModalReducer.data?.recordType,
  record_id: pluginState.openCommentModalReducer.data?.recordId,
  comments
})
```

File: webapp/src/containers/addOrViewComments/index.tsx [hunk 2]
```diff
@@ -77,31 +78,43 @@ const AddOrViewComments = () => {
     };
 
     useEffect(() => {
-        if (!pluginState.openCommentModalReducer.recordType || !pluginState.openCommentModalReducer.recordId) {
+        if (!pluginState.openCommentModalReducer.data?.recordType || !pluginState.openCommentModalReducer.data?.recordId) {
             return;
         }
 
         const payload: CommentsPayload = {
```
Comment (muazzamqazi) [inline webapp/src/containers/addOrViewComments/index.tsx:85]:
```suggestion
        const payload: CommentsPayload = getCommentsPayload();
```

File: webapp/src/containers/addOrViewComments/index.tsx [hunk 3]
```diff
@@ -124,8 +137,8 @@ const AddOrViewComments = () => {
...
```
Comment (muazzamqazi) [inline webapp/src/containers/addOrViewComments/index.tsx:139]:
```suggestion
       const payload: CommentsPayload = getCommentsPayload();
```

File: webapp/src/containers/addOrViewComments/index.tsx [hunk 4]
```diff
@@ -145,39 +158,53 @@ const AddOrViewComments = () => {
...
+                            {!apiError && <h4 className='comment-body__heading'>{Constants.CommentsHeading}</h4>}
+                            {commentsData ? (
+                                <>
+                                    <div className='comment-body__description-text'>{commentsData}</div>
+                                    {!showModalLoader && <p className='comment-body__footer'>{Constants.NoCommentsPresent}</p>}
+                                </>
+                            ) : (
+                                <>
+                                    {!showModalLoader && <p className='comment-body__footer'>{Constants.CommentsNotFound}</p>}
+                                </>
```
Comment (muazzamqazi) [inline webapp/src/containers/addOrViewComments/index.tsx:194]:
```suggestion
                                !showModalLoader && <p className='comment-body__footer'>{Constants.CommentsNotFound}</p>
```

File: webapp/src/containers/addOrViewComments/index.tsx [hunk 5]
```diff
@@ -46,6 +46,12 @@ const AddOrViewComments = () => {
         resetFieldStates();
     }, []);
 
+    const getCommentsPayload = (): CommentsPayload => ({
+        record_type: pluginState.openCommentModalReducer.data?.recordType,
+        record_id: pluginState.openCommentModalReducer.data?.recordId,
+        comments,
+    });
```
Comment (muazzamqazi) [inline webapp/src/containers/addOrViewComments/index.tsx:53]: Please check if we need to add comments for all payloads
```suggestion
    const getCommentsPayload = (includeComments: boolean): CommentsPayload => ({
        record_type: pluginState.openCommentModalReducer.data?.recordType,
        record_id: pluginState.openCommentModalReducer.data?.recordId,
        ...(includeComments ? {comments} : {}),
    });
```

**Review by muazzamqazi — APPROVED** [review]

File: webapp/src/containers/addOrViewComments/index.tsx (hunk 5 above)
Comment (ayusht2810) [reply webapp/src/containers/addOrViewComments/index.tsx:53]: I don't think we need to do this. We are sending the payload in `POST` request only in which we will need comments. In other requests (`GET`) we are using `record_type` and `record_id` in URL paths and comments are unused

File: webapp/src/containers/addOrViewComments/index.tsx [hunk 6]
```diff
@@ -77,31 +79,40 @@ const AddOrViewComments = () => {
...
+    }, [pluginState.openCommentModalReducer.data?.recordType, pluginState.openCommentModalReducer.data?.recordId]);
+
+    useEffect(() => {
+        const commentState = getCommentsState();
+        if (commentState.isLoading) {
+            setShowErrorPanel(false);
+        }
+
+        setShowModalLoader(commentState.isLoading);
+    }, [getCommentsState().isLoading]);
```
Comment (manojmalik20) [inline webapp/src/containers/addOrViewComments/index.tsx:99]: Can't you combine this `useEffect` with the below one?

File: webapp/src/containers/addOrViewComments/index.tsx [hunk 7]
```diff
@@ -123,11 +134,7 @@ const AddOrViewComments = () => {
             return;
         }
 
-        const payload: CommentsPayload = {
-            record_type: pluginState.openCommentModalReducer.recordType,
-            record_id: pluginState.openCommentModalReducer.recordId,
-        };
-
+        const payload = getCommentsPayload();
         setGetCommentsPayload(payload);
```
Comment (manojmalik20) [inline webapp/src/containers/addOrViewComments/index.tsx:138]: I think there's no use of this state variable.

File: webapp/src/containers/addOrViewComments/index.tsx [hunk 8]
```diff
@@ -1,7 +1,7 @@
 import React, {useCallback, useEffect, useState} from 'react';
 import {useDispatch} from 'react-redux';
 
-import {CircularLoader, CustomModal as Modal, ModalFooter, ModalHeader, ModalSubtitleAndError, TextField} from '@brightscout/mattermost-ui-library';
+import {CircularLoader, CustomModal as Modal, ModalFooter, ModalHeader, ModalSubtitleAndError, ResultPanel, TextField} from '@brightscout/mattermost-ui-library';
```
Comment (manojmalik20) [inline webapp/src/containers/addOrViewComments/index.tsx:4]:
```suggestion
import {CircularLoader, CustomModal as Modal, ModalFooter, ModalHeader, ModalSubtitleAndError, ResultPanel, TextArea} from '@brightscout/mattermost-ui-library';
```

File: webapp/src/types/common/index.d.ts [hunk 9]
```diff
@@ -61,6 +61,6 @@ type SubscriptionCardBody = {
 }
 
 type CommentModalData = {
-    recordType: RecordType;
+    recordType: RecordType | string;
```
Comment (manojmalik20) [inline webapp/src/types/common/index.d.ts:64]: Why this change?

File: webapp/src/containers/addOrViewComments/index.tsx (hunk 7 above)
Comment (ayusht2810) [reply webapp/src/containers/addOrViewComments/index.tsx:138]: already removed in this PR #83

File: webapp/src/containers/addOrViewComments/index.tsx (hunk 8 above)
Comment (ayusht2810) [reply webapp/src/containers/addOrViewComments/index.tsx:4]: this PR contains an older version of the mm UI library. It will be changed after getting in sync with PR #83

File: webapp/src/types/common/index.d.ts (hunk 9 above)
Comment (ayusht2810) [reply webapp/src/types/common/index.d.ts:64]: on closing modal we want to reset reducer states.... we were getting an error on setting `record_type` value to ''. so made this change.

Comment (manojmalik20) [reply webapp/src/types/common/index.d.ts:64]: What's the need for resetting the reducer states?

**Review by manojmalik20 — APPROVED** [review]
Remember to make appropriate changes when this PR will be synced with the latest code.

File: webapp/src/types/common/index.d.ts (hunk 9 above)
Comment (ayusht2810) [reply webapp/src/types/common/index.d.ts:64]: we are opening the modal on basis of whether the reducer states have these values present or not. and on closing the modal we are resetting them

Comment (manojmalik20) [reply webapp/src/types/common/index.d.ts:64]: Whenever the modal will open, these values will always be present because there's no way to open the modal except clicking on the Post action button. So, I don't think resetting the values is really necessary. What do you think?

Comment (ayusht2810) [reply webapp/src/types/common/index.d.ts]: updated the logic now

**Review by manojmalik20 — APPROVED** [review]
